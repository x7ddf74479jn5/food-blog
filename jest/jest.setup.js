// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
