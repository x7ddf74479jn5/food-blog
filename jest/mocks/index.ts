export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
