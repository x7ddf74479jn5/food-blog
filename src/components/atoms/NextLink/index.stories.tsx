import type { ComponentMeta, ComponentStory } from "@storybook/react";

import NextLink from ".";

export default {
  component: NextLink,
  title: "Atoms/NextLink",
} as ComponentMeta<typeof NextLink>;

const Template: ComponentStory<typeof NextLink> = (args) => <NextLink {...args}>Link</NextLink>;

export const Default = Template.bind({});
Default.args = {
  className: "text-2xl font-bold text-green-500",
  href: "",
};
