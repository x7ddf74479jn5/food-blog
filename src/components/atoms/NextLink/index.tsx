import Link from "next/link";

type Props = {
  href: string;
  className?: string;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  "aria-label"?: string;
  onClick?: () => void;
  children?: React.ReactNode;
  prefetch?: boolean;
};

const NextLink: React.FC<Props> = ({ children, href, prefetch = false, ...rest }) => {
  return (
    <Link href={href} prefetch={prefetch}>
      <a {...rest}>{children}</a>
    </Link>
  );
};

export default NextLink;
