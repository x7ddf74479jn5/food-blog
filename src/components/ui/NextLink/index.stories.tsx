import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import NextLink from ".";

export default {
  component: NextLink,
  title: "ui/NextLink",
} as ComponentMeta<typeof NextLink>;

export const Default: ComponentStoryObj<typeof NextLink> = {
  args: {
    className: "text-2xl font-bold text-green-500",
    href: "",
  },
};
