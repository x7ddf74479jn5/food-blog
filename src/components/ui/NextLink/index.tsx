import Link from "next/link";

type Props = {
  href: string;
  className?: string;
  "aria-label"?: string;
  onClick?: () => void;
  children?: React.ReactNode;
  prefetch?: boolean;
};

const NextLink: React.FC<Props> = ({ children, prefetch = false, ...rest }) => {
  return (
    <Link prefetch={prefetch} {...rest}>
      {children}
    </Link>
  );
};

export default NextLink;
