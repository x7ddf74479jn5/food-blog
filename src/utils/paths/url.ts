export const apiRoute = {
  apiPreview: "/api/preview",
  apiArticles: "/api/articles",
} as const;

export const urlTable = {
  home: "/",
  root: "/",
  articles: "/articles",
  categories: "/articles/categories",
  tags: "/articles/tags",
  preview: "/preview",
  search: "/search",
} as const;

const labelTable = {
  [urlTable.home]: "レシピ一覧へ",
  [urlTable.categories]: "カテゴリー別一覧へ",
  [urlTable.tags]: "タグ別一覧へ",
} as const;

type LabelTableKey = keyof typeof labelTable;

export const getBackLinks = (paths: LabelTableKey[]): { href: string; label: string }[] => {
  return paths.map((path) => {
    return { href: path, label: labelTable[path] };
  });
};
