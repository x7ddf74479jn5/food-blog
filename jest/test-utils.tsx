import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
