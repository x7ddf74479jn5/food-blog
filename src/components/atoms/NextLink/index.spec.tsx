import { render, screen } from "jest/test-utils";
import renderer from "react-test-renderer";

import { urlTable } from "@/utils/paths/url";

import NextLink from ".";

describe("components/atoms/NextLink", () => {
  const homeUrl = urlTable.home;
  it("snapshot", () => {
    const tree = renderer.create(<NextLink href={homeUrl}>Home</NextLink>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("OK: リンクが正しく表示されている", () => {
    render(<NextLink href={homeUrl}>Home</NextLink>);
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", homeUrl);
    expect(link).toHaveTextContent("Home");
  });
});
