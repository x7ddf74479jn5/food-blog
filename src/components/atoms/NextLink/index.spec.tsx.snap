// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/NextLink snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Home
</a>
`;
